### ISSUE TITLE
Provide detailed availability values at /bibliographic/:id/holding/:id/availability endpoint

### ISSUE THREAD

**Issue by tpendragon** [pulibrary/bibdata#937]
child of #573; follow-up to #1179

## Implementation

For this endpoint, we need to send item-level availability statuses. The status and status_label should be taken from:
1. work_order_type if it exists
2. else process_type if it exists
3. else base_status

`status` field should return "value"; `status_label` field should return "desc"

Note: An item will never have more than one `work_order_type` at once.

An example "at preservation" item:
MMS Id: 99116973763506421
Holding ID: 22245146270006421
Item ID: 23245146260006421

```
"base_status":
  {
    "value": "0",
    "desc": "Item not in place"
  }, 
“process_type”:
  {
    “value”: “WORK_ORDER_DEPARTMENT”,
    “desc”: “In Process”
  },
“work_order_type”:
  {
    “value”: “Pres”,
    “desc”: “At Preservation”
  },
 “work_order_at”:
  {
    “value”: “DEFAULT_CIRC_DESK-Pres”,
    “desc”: “Default Circulation Desk”
  },
```

## Background / Discussion

Previously we computed a wide variety of labels based on statuses that came out of voyager. We won't do that any longer. Below is discussion we had on the way to figuring out the new procedure.

Status labels [currently in marc liberation](https://github.com/pulibrary/marc_liberation/blob/090e90bf02cdbd80564c14bd9e8b01952f4a8a64/app/models/voyager_lookup.rb#L109-L146):
- [ ] 'All Items Available' (Now `more_items == true && status.start_with?('on-site')`)
- [ ] 'Available' (Now `['Not Charged', 'On Shelf']`)
- [ ] 'Long overdue' (Now `['Lost--system applied', 'On-site - lost--system applied']`)
- [ ] 'Lost' (Now `['Lost--library applied', 'On-site - lost--library applied']`)
- [ ] Returned (Now `['Discharged']`)
- [ ] In transit (Now `['In transit discharged']`)
- [ ] In process (Now `['In process', 'On-site - in process']`)
- [ ] Checked out (Now `['Charged', 'Renewed', 'Overdue', 'On hold', 'In transit', 'In transit on hold', 'In transit discharged', 'At bindery',  'Remote storage request', 'Hold request', 'Recall request']`)
- [ ] Missing (Now `['Missing', 'Claims returned', 'Withdrawn', 'On-site - missing', 'On-site - claims returned', 'On-site - withdrawn']`)
- [ ] See front desk (Starts with 'on site - ')
- [ ] On-site access ('on-site')
- [ ] Pending order ('Pending Order')
- [ ] Order received ('order received')
- [ ] On-order ('on-order') - see #630 for alma details
- [ ] ? (There's a fallback to whatever Voyager has if the above doesn't fit)

### Alma implementation

There is a list of process types at https://developers.exlibrisgroup.com/alma/apis/docs/xsd/rest_item.xsd/. The list shows demo process types but is supposed to show your institution's actual process types if you log in. The only type of process_type our records currently have is "technical". Everything with one of those statuses will have availability "not available".

The records are migrated with process_type "technical - migration" and the voyager data in local_notes_3. pul staff will manually move them into the various other process types in batches. This means they will create a "work order" for each process type and then create statuses for that work order.

We should be able to find fixtures for items with the checked out status. These will have a process_type "loan".

We can also create a fixture for an item with a process type of "technical - migration". The next step will be converting a handful of these to use real work orders / process types.

#### Impact on Requests app

3 Broad categories of status values (not status label values!), as far as Requests is concerned. Behavior culminates in the [requestable form](https://github.com/pulibrary/requests/blob/main/app/views/requests/request/_requestable_form.html.erb).
- Charged or missing (`borrow_direct?`)
  - user is prompted to check borrow direct
- On-order or Pending (`request?`)
  - In-process and on-order items are sent through an email workflow so catalogers can bump up its priority. 
- Available (requests logic is that it's available if it's not in the other categories) (`pick_up?`)
  - user can schedule a pickup.

related migration ticket in requests: https://github.com/pulibrary/requests/issues/891

**Comment by mzelesky** [comment]
This really cannot be done until January. Statuses in Alma are completely different from statuses in Voyager, and it will take a while to get examples of them. This will have to be done in early January.

**Comment by tpendragon** [comment]
Thanks @mzelesky! Marking this issue as blocked.

**Comment by mzelesky** [comment]
Some documentation on work orders in Alma (which lead to process types):

https://knowledge.exlibrisgroup.com/Alma/Training/Alma_Essentials/Alma_Essentials_-_English/E_Resource_Management/05_Work_Orders

https://knowledge.exlibrisgroup.com/Alma/Product_Documentation/010Alma_Online_Help_(English)/030Fulfillment/Resource_Requests/020Managing_Requests_and_Work_Orders

https://knowledge.exlibrisgroup.com/Alma/Product_Documentation/010Alma_Online_Help_(English)/050Administration/050Configuring_General_Alma_Functions/040Work_Orders_and_Departments#Adding_a_Work_Order_Type_Status

**Comment by mzelesky** [comment]
This item is in a Preservation process:

MMS Id: 99116973763506421
Holding ID: 22245146270006421
Item ID: 23245146260006421
Process type: Pres

**Comment by hectorcorrea** [comment]
It looks like the `process_type` values available for our installation are are:

```
ACQ,                        "Acquisition"
CLAIM_RETURNED_LOAN,        "Claimed Returned"
HOLDSHELF,                  "Hold Shelf"
ILL,                        "Resource Sharing Request"
LOAN,                       "Loan"
LOST_ILL,                   "Lost Resource Sharing Item"
LOST_LOAN,                  "Lost"
LOST_LOAN_AND_PAID,         "Lost and paid"
MISSING,                    "Missing"
TECHNICAL,                  "Technical - Migration"
REQUESTED,                  "Requested"
TRANSIT,                    "Transit"
TRANSIT_TO_REMOTE_STORAGE,  "In Transit to Remote Storage"
WORK_ORDER_DEPARTMENT,      "In Process"
```

Fetched via https://api-na.hosted.exlibrisgroup.com/almaws/v1/conf/code-tables/PROCESSTYPE?apikey=enter-your-key

**Comment by hackartisan** [comment]
Yeah I'm still not really sure how process_type is related here. I think what we want is work_order_type. But we may need these, too, at some point?

**Comment by mzelesky** [comment]
I _think_ process_types include things that aren't on work orders. For example, HOLDSHELF is just a status that gets toggled without being associated with a work order. Work orders all have a process_type of WORK_ORDER_DEPARTMENT.

**Comment by hackartisan** [comment]
@mzelesky Do you think the logic is: If `process_type` is `WORK_ORDER_DEPARTMENT` use the `work_order_type` as the `status` otherwise use `process_type` as the `status`?

**Comment by hectorcorrea** [comment]
Closed via PR #1275. Individual issues can now be traced via separate tickets.
